Sources/CoreLocationMiddleware/CoreLocationMiddleware.swift

```diff
@@ -114,7 +159,15 @@ public final class CoreLocationMiddleware: Middleware {
     }
     
     func locationManager(_ manager: CLLocationManager, didFailWithError error: Error) {
-        output?.dispatch(.receiveError(error))
+        output?.dispatch(.receiveError(error as! CLError))
```

Comment:
Force cast makes me uncomfortable. But I understand that there are two groups of people, the ones that like crash and the ones that don't :)

Reply:
I actually don't like it either ! Maybe I should use the conditional form of the downcast... and just return a made up error in case the downcast doesn't work. I'll take a stab at it.

Reply:
You can also create your own error type
```swift
enum CoreLocationError {
    case knownType(CLError)
    case unknownError(Error)
}
```

Or if this is too much, just send the Error as you got it and the customer decides what to do with it.

Reply:
I was trying to make `LocationAction` conform to `Equatable`. But, of course, `Error` can be made `Equatable` via an extension.
So I guess if we can live with non-conformance to `Equatable` we should be fine either way. But if it's desired, or required, then we can't have `Error` in there.
Any recommendation on `Equatable` ?

Reply:
I guess there just really is no need for it to be Equatable (as opposed to State).

Reply:
Actions don't usually have to be equatable, only State. But if you want to keep "lastError" on State, then you have to deal with the fact that Error can't be made Equatable.
What I often do, when I face that issue, is making a custom `static func ==`, comparing all properties and when I get into the Error property I compare the localizedError String. It's a "meh" solution, but it works for my scenarios. Mirror could be an option, but it's expensive (reflection etc).

There's also a way of making a PropertyWrapper for custom equatable of a single type... You could make a `@EquatableError var error: Error` and your `EquatableError: Equatable`'s `==` will hold the logic to compare localizedError property.

Reply:
This is somewhat similar to what I am doing in [CoreLocation-Redux](https://github.com/npvisual/CoreLocation-Redux/blob/b1c37eee3e9daf5616ebf3f6c7cc2b73f2ac80c4/Shared/ReduxAppModule.swift#L179), except I am using  `localizedDescription` and storing error as a String (so it's already equatable) : 

```
extension Reducer where ActionType == CoreLocationAction, StateType == AppState {
    static let location = Reducer { action, state in
        var state = state
        switch action {
        ....
        case let .triggerError(error):
            state.error = error.localizedDescription
```

So I think I will remove the Equatable requirement from LocationAction and just pass `Error`. But please see additional comment regarding moving returned data into State as opposed to via Action.

Reply:
See #7 to further discuss.